tdp/core/runner/deployment_plan.py

```diff
@@ -65,47 +75,54 @@ def from_operations(operations):
-                deployment_log_to_resume.filter_expression,
-                deployment_log_to_resume.filter_type,
-            )
-        elif deployment_log_to_resume.deployment_type == DeploymentTypeEnum.OPERATIONS:
-            raise Exception(
-                f"Resuming from an Operations deployment is not yet supported"
+    def from_failed_deployment(dag, deployment_log):
+        if deployment_log.state == StateEnum.SUCCESS:
+            raise NothingToResumeError(
+                f"Nothing to resume, deployment #{deployment_log.id} was successful"
             )
-        elif deployment_log_to_resume.deployment_type == DeploymentTypeEnum.RESUME:
-            raise Exception(f"Resuming from an Resume deployment is not yet supported")
 
-        if deployment_log_to_resume.state == StateEnum.SUCCESS:
-            raise NothingToResumeError(
-                f"Nothing to resume, deployment #{deployment_log_to_resume.id} was successful"
+        if deployment_log.deployment_type == DeploymentTypeEnum.DAG:
+            original_operations = DeploymentPlan.from_dag(
+                dag,
+                deployment_log.targets,
+                deployment_log.sources,
+                deployment_log.filter_expression,
+                deployment_log.filter_type,
+                deployment_log.restart,
+            ).operations
+            original_operation_names = [
+                operation.name for operation in original_operations
+            ]
+        elif deployment_log.deployment_type in (
+            DeploymentTypeEnum.OPERATIONS,
+            DeploymentTypeEnum.RESUME,
+        ):
+            original_operation_names = deployment_log.targets
+            original_operations = [
+                dag.collections.operations[name] for name in original_operation_names
+            ]
+        else:
+            raise UnsupportedDeploymentTypeError(
+                f"Resuming from a {deployment_log.deployment_type} is not supported"
             )
 
-        original_operations = [
-            operation.name for operation in deployment_plan_to_resume.operations
-        ]
-        succeeded_operations = [
-            operation.operation
-            for operation in deployment_log_to_resume.operations
-            if operation.state == StateEnum.SUCCESS
-        ]
-        remaining_operations_list = list(
-            set(original_operations) - set(succeeded_operations)
-        )
-        remaining_operations = [
-            operation
-            for operation in deployment_plan_to_resume.operations
-            if operation.name in remaining_operations_list
-        ]
+        if len(deployment_log.operations) > 0:
+            last_operation_log = deployment_log.operations[-1]
+            try:
+                index = original_operation_names.index(last_operation_log.operation)
+            except ValueError as e:
+                raise GeneratedDeploymentPlanMissesOperationError(
```

Comment:
This is meant to fail when we are trying to resume a failed operation with a different version of the dag